src/server/scene/basic_idle_hub.h

```diff
@@ -76,6 +88,7 @@ class BasicIdleHub : public IdleHub
 
     std::shared_ptr<time::Clock> const clock;
     std::unique_ptr<time::Alarm> const alarm;
+    std::weak_ptr<BasicIdleHub::WakeLock> wake_lock;
```

Comment:
Generally we put mutable state below the `mutex` to indicate it should only be changed under lock.